app/upgrades/v17/upgrades_test.go

```diff
@@ -125,10 +142,27 @@ func (suite *UpgradeTestSuite) TestUpgrade() {
 					lastPoolID = poolID
 				}
 
-				return expectedCoinsUsedInUpgradeHandler, lastPoolID
+				existingPool := suite.PrepareConcentratedPoolWithCoins("ibc/1480B8FD20AD5FCAE81EA87584D269547DD4D436843C1D20F15E00EB64743EF4", "uosmo")
+
+				// create few TWAP records for the pools
+				t1 := dummyTwapRecord(existingPool.GetId(), time.Now().Add(-time.Hour*24), "ibc/1480B8FD20AD5FCAE81EA87584D269547DD4D436843C1D20F15E00EB64743EF4", "uosmo", sdk.NewDec(10),
+					sdk.OneDec().MulInt64(10*10),
+					sdk.OneDec().MulInt64(3),
+					sdk.ZeroDec())
+
+				suite.App.TwapKeeper.StoreNewRecord(suite.Ctx, t1)
+
+				clPoolTwapRecordPreUpgrade, err := keepers.TwapKeeper.GetAllMostRecentRecordsForPool(ctx, existingPool.GetId())
+				suite.Require().NoError(err)
+
+				clPoolTwapRecordHistoricalPoolIndex, err := keepers.TwapKeeper.GetAllHistoricalPoolIndexedTWAPsForPoolId(ctx, existingPool.GetId())
+				suite.Require().NoError(err)
```

Comment:
Its good that we added this here, but we need to add logic in the success cases that checks the new record post upgrade to ensure both records were changed as expected. I would also create more than one historical record. Then, after the upgrade, you can iterate over the historical records to see if the values were swapped, then just compare the current record to ensure that flipped as well.

Reply:
oh yea sorry i had this test created but forgot to push https://github.com/osmosis-labs/osmosis/pull/5939/commits/78c7cec1ef24b264211f31a9440f5bd03025910c#diff-8d0951a4357ce3ef3f57168835dcc72078bd30337d8f9882c3b0d1e64a815e77R180-R195

Reply:
lmk if this suffice, i can write more test too!

Reply:
added more detailed test here: https://github.com/osmosis-labs/osmosis/pull/5939/commits/4ea33fa8248b92b65c17dc271335010d91716985